src/OrleansProviders/Streams/Common/PooledCache/ICacheDataAdapter.cs

```diff
@@ -59,6 +60,24 @@ public interface ICacheDataAdapter<in TQueueMessage, TCachedMessage>
         /// Assignable purge action.  This is called when a purge request is triggered.
         /// </summary>
         Action<IDisposable> PurgeAction { set; }
+
+        /// <summary>
+        /// Handler for when a cachedMessage got purged
+        /// </summary>
+        /// <param name="cachedMessage"></param>
+        void SignalMessagePurged(ref TCachedMessage cachedMessage);
+
+        /// <summary>
+        /// Determine action to make on a purgeRequest
+        /// </summary>
+        /// <param name="purgeRequest"></param>
+        void ConsultPurgeRequest(IDisposable purgeRequest);
```

Comment:
I have absolutely zero idea of what this is supposed to do, by looking at the method name or comments.